src/authn/noop/mod.rs

```diff
@@ -0,0 +1,15 @@
+use super::Authenticator;
+use crate::authz::noop::Anonymous;
+
+/// An authenticator that simply returns an anonymous user
+#[derive(Clone, Debug)]
+pub struct NoopAuthenticator;
```

Comment:
This all looks great! Just a small optionally ignore-able (at your discretion) nit: Maybe instead of Noop, you call it Always or Unconstrained. This might just be me, but a no-op in the context of authentication isn't immediately obvious.

Reply:
This is fair. I'm all for bikeshedding a name if it adds to understanding. I think I'll change it to `Always`